nautobot_circuit_maintenance/handle_notifications/handler.py

```diff
@@ -200,21 +202,22 @@ def create_raw_notification(logger: Job, notification: MaintenanceNotification,
     """
     # Insert raw notification in DB even failed parsing
     try:
-        raw_entry, created = RawNotification.objects.get_or_create(
+        raw_entry = RawNotification(
             subject=notification.subject,
             provider=provider,
             raw=notification.raw_payload,
             sender=notification.sender,
             source=NotificationSource.objects.filter(name=notification.source).last(),
+            date=parser.parse(notification.date),
         )
-    except Exception as exc:
-        logger.log_warning(message=f"Raw notification '{notification.subject}' not created because {str(exc)}")
-        return None
-
-    if not created:
+        raw_entry.save()
+    except IntegrityError as exc:
```

Comment:
Doesn't `IntegrityError` generally leave the DB in a bad state? Are we able to safely continue processing additional notifications in the case where this happens? In the previous (existing) version of the code, I saw things like this:

![image](https://user-images.githubusercontent.com/5603551/133826096-3679d3c4-d28e-4e89-9e12-b00da442d4ec.png)

Reply:
My idea capturing the `IntegrityError` was to deal with it and do not touch the database with creating references (for instance with the `CircuitMaintenance`), as it was happening before with the `get_or_create` that was evaluated into the DB later in the process (and the other entities where also "created"). 
Now, as soon as we detect the duplicated entry in the DB, we stop processing and do not create any reference, so we avoid bad state.
If this approach looks flaky, another alternative is to check for an instance with the `unique_together` elements before creating. I thought this was "simpler", but open to suggestions.
I was able to reproduce the issue you are mentioning in my dev environment, and this  fixed the issue for me and also for the new test case added.

Reply:
OK, if you have been able to reproduce the issue and verify that this fixes it, I'm OK with this approach. Thanks!